src/AutoRest.CSharp/Mgmt/Generation/MgmtClientBaseWriter.cs

```diff
@@ -254,11 +254,55 @@ protected virtual void WriteChildResourceEntries()
                     {
                         WriteResourceCollectionEntry(resource.ResourceCollection, signature);
                     }
+
+                    if (!(This is MgmtExtensionClient)) // we don't need to generate `Get{Resource}` methods in ExtensionClient
+                    {
+                        WriteChildResourceGetMethod(resource.ResourceCollection, true);
+                        WriteChildResourceGetMethod(resource.ResourceCollection, false);
+                    }
```

Comment:
Now this loop are written in different layers - is it possible that we refactor the part that writes the "return collection" also to wrap it into a function so that this loop is more readable